NineChronicles.Headless.Tests/GraphTypes/States/Models/ShopStateTypeTest.cs

```diff
@@ -42,5 +44,109 @@ public async Task Query()
                 queryResult.Data
             );
         }
+
+        [Theory]
+        [MemberData(nameof(Members))]
+        public async Task QueryWithArguments(int? id, string itemSubType, int? price, int expected)
+        {
+            var queryArgs = $"id: {id}";
+            if (!(itemSubType is null))
+            {
+                queryArgs += $", itemSubType: {itemSubType}";
+            }
+
+            if (!(price is null))
+            {
+                queryArgs += $", maximumPrice: {price}";
+            }
+            var query = $@"query {{
+                products({queryArgs}) {{
+                    sellerAgentAddress
+                    sellerAvatarAddress
+                    price
+                    itemUsable {{
+                        itemId
+                        itemType
+                        itemSubType
+                    }}
+                    costume {{
+                        itemId
+                        itemType
+                        itemSubType
+                    }}
+                }}
+            }}";
+            var queryResult = await ExecuteQueryAsync<ShopStateType>(query, source: Fixtures.ShopStateFX());
+            var products = queryResult.Data.As<Dictionary<string, object>>()["products"].As<List<object>>();
+            Assert.Equal(expected, products.Count);
+        }
+
+        [Theory]
+        [InlineData("id: 1.0")]
+        [InlineData("maximumPrice: 1.1")]
+        [InlineData("itemSubType: Costume")]
+        public async Task QueryWithInvalidArguments(string queryArgs)
+        {
+            var query = $@"{{
+                address
+                products({queryArgs}) {{
+                    sellerAgentAddress
+                    sellerAvatarAddress
+                    price
+                    itemUsable {{
+                        itemId
+                        itemType
+                        itemSubType
+                    }}
+                    costume {{
+                        itemId
+                        itemType
+                        itemSubType
+                    }}
+                }}
+            }}";
+            var queryResult = await ExecuteQueryAsync<ShopStateType>(query, source: new ShopState());
+            Assert.Null(queryResult.Data);
+            Assert.Equal("ARGUMENTS_OF_CORRECT_TYPE", queryResult.Errors.First().Code);
```

Comment:
If you use `IEnumerable.First()` and check one, how about check also it has only one error with `Assert.Single()`?